src/newgrf_text.cpp

```diff
@@ -767,41 +768,42 @@ static void HandleNewGRFStringControlCodes(std::string_view str, TextRefStack &s
  * @param stack The TextRefStack.
  * @param[out] params Output parameters
  */
-static void ProcessNewGRFStringControlCode(char32_t scc, const char *&str, TextRefStack &stack, std::vector<StringParameter> &params)
+static void ProcessNewGRFStringControlCode(char32_t scc, StringConsumer &str, TextRefStack &stack, std::vector<StringParameter> &params)
 {
 	/* There is data on the NewGRF text stack, and we want to move them to OpenTTD's string stack.
 	 * After this call, a new call is made with `modify_parameters` set to false when the string is finally formatted. */
 	switch (scc) {
 		default: return;
 
 		case SCC_PLURAL_LIST:
-			++str; // plural form
+			(void)str.Uint8Consume(); // plural form
```

Comment:
```suggestion
			static_cast<void>(str.Uint8Consume()); // plural form
```
Also for the other cases that got introduced, given we are trying to move away from C-style casts.

Or maybe implement a skip function?